### DISCUSSION TITLE
Could not found the root route (GET /  404 Not Found  ) on Dev

### DISCUSSION THREAD

**Discussion by emdadul38** [kinngh/shopify-node-express-mongodb-app#56] [General]
Thanks for creating the Shopify app template. I'm facing an error while installing the app in dev mode, but everything is okay on production mode. 

![Screenshot from 2023-02-21 18-06-51](https://user-images.githubusercontent.com/10665798/220340657-017eefe3-ceb2-45d9-8c1f-b95bf7ae2d93.png)

**Comment by kinngh** [comment]
In Dev mode, use the ngrok url install method available in your `.env` instead of using Shopify partners page. This is because Vite and Express are being served from two different ports and running Vite directly skips middleware checks from Express, causing the app to not check for auth before showing the UI. `This is a dev mode only bug and is currently being worked on^`

Ensure when in dev mode, you're running `npm run ngrok` for port `5173` and your `.env` has no `PORT` assigned. If you're using `ngrok` from your terminal, use `ngrok http 5173` instead.

PS: This should have been an issue, not discussion^

**Reply by emdadul38** [reply]
Thanks. It's working

**Reply by shraddhadev24** [reply]
I am on same issue, @kinngh so we need to run the ngrok for 5173 & 8081 seperately ? And where to put the ngrok 5173  url ?

**Reply by kinngh** [reply]
@shraddhadev24 when you're in dev mode, have 2 terminal windows open.

1.` npm run ngrok`
2. `npm run dev`

When you want to run in production mode, first `npm run build` and then
1. `npm run ngrok:prod`
2. `npm run start`

In both the cases, after you run `npm run ngrok` / `npm run ngrok:prod`, you need to run `npm run update:url` so it updates the URLs. Make sure you update the Ngrok URLs so url updates runs properly.

**Reply by shraddhadev24** [reply]
By using this are we bypassing the app_installation in dev mode ?

**Reply by kinngh** [reply]
> By using this are we bypassing the app_installation in dev mode ?

Can you explain better please? You need to give me more context for it to make sense.

**Reply by shraddhadev24** [reply]
Sure , my bad

I followed these steps to run on DEV_MODE

1. npm run ngrok, use this URL in .env file
2. npm run update:url
3. npm run dev

After this installation doesn't work, but if I run ngrok on 8081 and put that url in .env installation happen finely just I am getting /GET error for the UI on DEV MODE . So I just want to understand as /auth /auth/tokens  are in server side code if in .env SHOPIFY_APP_URL  = we put vite app url how the backend is getting served to shopify, how the installation call will happen from shopify to the server.

Does I made it clear ?

@kinngh by the way A BIG THANK YOU, this repo is literally a best resource over internet to start developing a shopify app for a begineer shopify developer.

**Reply by kinngh** [reply]
Yes there's been an issue with even my understanding of how Vite works, but now that I've had the time to go over a whole lotta stuff, I'll be deploying a fix mid-august that rewrites the whole Vite fiasco and fixes it. If I'm being really honest, don't break your head over what's happening with ports. 

In dev mode, you're doing it right. Start ngrok -> update your env -> run update:url -> run dev. If you click on `Install` from your partner dashboard, it won't work. BUT if you head over to `https://ngrok.url.com/auth?shop=shopname.myshopify.com`, it'll install it and then you're good to go. This doesn't happen in production mode because the build step compiles the entire React app into static files and is then servable via Express, which is why in `server/index.js` you'd find the code where it's serving the static `index.html` file.

Again, just to put emphasis on installation, manually create your install URL and you'll be good to go^

**Reply by Ajeetesh1999** [reply]
@kinngh still after following the above steps, i am facing the same issue, watch this once
![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/78473901/e7638f15-6ce2-4c87-96ed-bfa03cb6ed3c)

**Reply by kinngh** [reply]
@Ajeetesh1999 yhe screenshot adds no context, can you please be more specific so I can see what's going wrong?

**Reply by Ajeetesh1999** [reply]
@kinngh the app installed successfully the the screen show this error
![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/78473901/65b4b108-ae14-4d04-a2a2-9b528c36e7ad)
i checked on three different store, but the same issue is occuring

**Reply by kinngh** [reply]
@Ajeetesh1999 ok you need to he more clear. What commands are you running?
